### PR TITLE
Fix folly CXX_STD

### PR DESCRIPTION
**Pull request by Hopobcn**
Specify library name and version:  **folly/all**

* Fixes https://github.com/conan-io/conan-center-index/issues/11937
* updates `fmt` dependency to latest 7.x.x supported in CCI (7.1.3)
* Migrate test_package to comply with v2 linter 

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by Hopobcn** [comment]
`recipes/folly/all/conanfile.py:144: [E1101(no-member), FollyConan._configure_cmake] Module 'conans.tools' has no 'cross_building' member`

Mmm even if the recipe doesn't use `cross_building` this linter check is failing. This means now it is forbidden to include `conans.tools` ?

CC: @SSE4

**Comment by Hopobcn** [comment]
I'm not sure c3i bot that parses test_package is working well. Waiting for SSE4 feedback but probably I will rollback [6a33810](https://github.com/conan-io/conan-center-index/pull/11938/commits/6a33810821f68872ced0a5564239dfad026d9ce9)

**Comment by SSE4** [comment]
> I'm not sure c3i bot that parses test_package is working well. Waiting for SSE4 feedback but probably I will rollback [6a33810](https://github.com/conan-io/conan-center-index/pull/11938/commits/6a33810821f68872ced0a5564239dfad026d9ce9)

it raies for this line (not for test_package): https://github.com/conan-io/conan-center-index/pull/11938/files#diff-c76b59bc54e55f5a4d20622555a9653f5345015c0c85716862cdab5139eb4dc6R144
P.S. no need to revert, it doesn't block the merge

**Review by prince-chrismc — DISMISSED** [review]

**Comment by prince-chrismc** [comment]
No obvious reason why it failed 🤔

**Comment by uilianries** [comment]
There is a conflict version: `   WARN: libevent/2.1.12: requirement openssl/1.1.1m overridden by folly/2019.10.21.00 to openssl/1.1.1q `

**Comment by uilianries** [comment]
It seems be a runtime error when running the test_package.

**Comment by prince-chrismc** [comment]
```
libunwind/1.5.0: Already installed!
folly/2019.10.21.00: Already installed!
folly/2019.10.21.00 (test package): Generator 'CMakeDeps' calling 'generate()'
folly/2019.10.21.00 (test package): Generator 'VirtualRunEnv' calling 'generate()'
folly/2019.10.21.00 (test package): Generator txt created conanbuildinfo.txt
folly/2019.10.21.00 (test package): Calling generate()
folly/2019.10.21.00 (test package): WARN: Using the new toolchains and generators without specifying a build profile (e.g: -pr:b=default) is discouraged and might cause failures and unexpected behavior
folly/2019.10.21.00 (test package): WARN: Using the new toolchains and generators without specifying a build profile (e.g: -pr:b=default) is discouraged and might cause failures and unexpected behavior
ERROR: folly/2019.10.21.00 (test package): Error in generate() method, line 18
	tc.variables["FOLLY_VERSION"] = self.dependencies["fmt"].ref.version
	KeyError: {'ref': fmt/unknown@unknown/unknown, 'build': False, 'direct': True, 'test': False, 'visible': True}
********************************************************************************
conan test conan-center-index/recipes/folly/all/test_v1_package/conanfile.py folly/2019.10.21.00@#c34697b5055a561723884d696ec459f8
```

File: recipes/folly/all/test_package/conanfile.py [hunk 1]
```diff
@@ -1,18 +1,31 @@
...
     settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake", "cmake_find_package_multi"
+    generators = "CMakeDeps", "VirtualRunEnv"
+
+    def requirements(self):
+        self.requires(self.tested_reference_str)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["FOLLY_VERSION"] = self.dependencies["fmt"].ref.version
```
Comment (uilianries) [inline recipes/folly/all/test_package/conanfile.py:18]:
```suggestion
```
I don't see any usage in test_package/CMakeLists.txt. Also, you tried to capture `fmt` version, which should be `folly` instead. Conan 2 is a bit tricky when exposing transitive dependencies, it only exposes what you explicitly added as requirement.

Comment (Hopobcn) [reply recipes/folly/all/test_package/conanfile.py:18]: Yup I think I forgot to add this in `test_package/CMakeLists.txt`

```
if (${FOLLY_VERSION} VERSION_LESS "2021.07.20.00")
    set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 14)
else()
    set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 17)
endif()
```

> Also, you tried to capture fmt version, which should be folly instead

uff ctl+c/v mishap... I will change it.

Comment (Hopobcn) [reply recipes/folly/all/test_package/conanfile.py]: Ah no, the code was there but github doesn't show the 'unchanged lines' until you click the "Expand down" button (fail :-P).

I have fixed the self.dependencies .. should be working now.

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**